src/components/interface/StepsPanel.tsx

```diff
@@ -1,55 +1,133 @@
-import { Badge, Group, Stack } from '@mantine/core';
+import { NavLink, Text } from '@mantine/core';
+import { ReactNode, useMemo } from 'react';
 import { useNavigate, useParams } from 'react-router-dom';
 import { IconArrowsShuffle } from '@tabler/icons-react';
+import { useStudyConfig } from '../../store/hooks/useStudyConfig';
+import { deepCopy } from '../../utils/deepCopy';
 import { useCurrentStep } from '../../routes';
 import { OrderObject } from '../../parser/types';
 
-export function StepsPanel({ order }: { order: OrderObject }) {
+const getFlatMap = (orderObj: OrderObject): string[] => orderObj.components.flatMap((component) => (typeof component === 'string' ? component : getFlatMap(component)));
+
+const getVisibleChildComponent = (
+  sequence: string[],
+  orderObj: OrderObject,
+) => {
+  const flatObj = getFlatMap(orderObj);
+
+  const visibleChild = flatObj.find((component) => sequence.indexOf(component) !== -1);
+
+  return visibleChild;
+};
```

Comment:
Re-write as with function keyword

```suggestion
function getFlatMap (orderObj: OrderObject){
  return orderObj.components.flatMap((component) => (typeof component === 'string' ? component : getFlatMap(component)));
}

function getVisibleChildComponent(
  sequence: string[],
  orderObj: OrderObject,
){
  const flatObj = getFlatMap(orderObj);

  const visibleChild = flatObj.find((component) => sequence.indexOf(component) !== -1);

  return visibleChild;
}
```